pkg/daemon/update.go

```diff
@@ -598,6 +682,20 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 
 	klog.Infof("Checking Reconcilable for config %v to %v", oldConfigName, newConfigName)
 
+	err = upgrademonitor.GenerateAndApplyMachineConfigNodes(
+		&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePrepared, Reason: string(mcfgalphav1.MachineConfigNodeUpdateCompatible), Message: fmt.Sprintf("comparing MachineConfigs as part of the preparation phase")},
```

Comment:
similar to above, we don't have to collapse all the (very fast) scenarios, but it might be good to reduce some API calls here and there.

Basically, if we succeed, the state will transition almost instantly, and if we fail the degrade will cover it, in scenarios like this

Reply:
I agree, and I think joel shares some of this sentiment. I think we wanted to use the Unknown/T/F stuff for common error points, so I will switch some of these around